modules/renter/upload.go

```diff
@@ -102,6 +103,10 @@ func (r *Renter) Upload(up modules.FileUploadParams) error {
 	lockID := r.mu.Lock()
 	defer r.mu.Unlock(lockID)
 
+	if filepath.Ext(up.Filename) != filepath.Ext(up.Nickname) {
+		return errors.New("nickname and file name must have the same extension")
+	}
```

Comment:
I guess it is a frontend thing. If you add it to the front I'll remove it from the back.

It's one of those sharp-corner things. It makes it easier to shoot yourself in the foot and share a file where nobody will know the file extension they are supposed to use when downloading.